### PR TITLE
Feature add bookmarks fx 2

### PR DESCRIPTION
**Pull request by alextakada322**
I think I understand it better now youve broken it down haha, take a look at let me know

### CODE REVIEW

**Review by jakekaplan95 — APPROVED** [review]

File: src/components/PublicBookmarks.js [hunk 1]
```diff
@@ -1,21 +1,61 @@
...
-        <VerticalFlexBox width='40%' alignItems='center'>
-            <SubTitle>
-                Explore
-            </SubTitle>
+
+function PublicBookmarks(props){
+    const [bookmarks, setBookmarks] = useState(null);
+
+    const URL = "https://jab-bookmarkd-backend.herokuapp.com/bookmarks"
+
```
Comment (BlakeEriks) [inline src/components/PublicBookmarks.js:12]: const bookmarkService = useBookmarks()

File: src/components/PublicBookmarks.js [hunk 2]
```diff
@@ -1,21 +1,61 @@
...
+    const getBookmarks = async () => {
+        const response = await fetch(URL);
```
Comment (BlakeEriks) [inline src/components/PublicBookmarks.js:14]: const response = await bookmarkService.explore()

File: src/components/PublicBookmarks.js [hunk 3]
```diff
@@ -1,21 +1,61 @@
...
 
-const PublicBookmarks = () => {
-    return (
-        <VerticalFlexBox width='40%' alignItems='center'>
-            <SubTitle>
-                Explore
-            </SubTitle>
+
+function PublicBookmarks(props){
+    const [bookmarks, setBookmarks] = useState(null);
```
Comment (BlakeEriks) [inline src/components/PublicBookmarks.js:9]: useState([])

File: src/components/PublicBookmarks.js [hunk 4]
```diff
@@ -1,21 +1,61 @@
...
+
+    const URL = "https://jab-bookmarkd-backend.herokuapp.com/bookmarks"
+
+    const getBookmarks = async () => {
+        const response = await fetch(URL);
+        const data = await response.json();
```
Comment (BlakeEriks) [inline src/components/PublicBookmarks.js:15]: we dont need this

File: src/components/PublicBookmarks.js [hunk 5]
```diff
@@ -1,21 +1,61 @@
...
+        setBookmarks(data);
```
Comment (BlakeEriks) [inline src/components/PublicBookmarks.js:16]: setBookmarks(response)

File: src/components/PublicBookmarks.js [hunk 6]
```diff
@@ -1,21 +1,61 @@
...
+
+    const createBookmarks = async (bookmarks) => {
+    // make post request to create bookmarks
+    await fetch(URL, {
+      method: "post",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(bookmarks),
+    });
```
Comment (BlakeEriks) [inline src/components/PublicBookmarks.js:27]: const res = await bookmarkService.create(bookmark)

File: src/components/PublicBookmarks.js [hunk 7]
```diff
@@ -1,21 +1,61 @@
...
+    <main>
+        <Switch>
+            <VerticalFlexBox width='40%' alignItems='center'>
+            <h2>Explore</h2>
+            <hr />
+            <Route exact path="/">
+                <Index bookmarks={bookmarks} createBookmarks={createBookmarks} />
+            </Route>
             <ul>
                 <Bookmark website={'Google'} />
```
Comment (BlakeEriks) [inline src/components/PublicBookmarks.js:46]: createBookmark={createBookmark}

File: src/components/PublicBookmarks.js [hunk 8]
```diff
@@ -1,21 +1,61 @@
...
+
+return (
+    <main>
+        <Switch>
+            <VerticalFlexBox width='40%' alignItems='center'>
+            <h2>Explore</h2>
+            <hr />
+            <Route exact path="/">
+                <Index bookmarks={bookmarks} createBookmarks={createBookmarks} />
+            </Route>
```
Comment (BlakeEriks) [inline src/components/PublicBookmarks.js:44]: We don't need a new route.

File: src/components/PublicBookmarks.js [hunk 9]
```diff
@@ -1,21 +1,61 @@
...
             <ul>
```
Comment (BlakeEriks) [inline src/components/PublicBookmarks.js:45]: {bookmarks.map( bookmark => <Bookmark key={bookmark._id} {...bookmark} />)}

File: src/components/PublicBookmarks.js [hunk 10]
```diff
@@ -1,21 +1,61 @@
...
                 <Bookmark website={'Google'} />
                 <Bookmark website={'Facebook'} />
                 <Bookmark website={'Amazon'} />
                 <Bookmark website={'MDN'} />
             </ul>
-        </VerticalFlexBox>
+            
+            <hr />
+            <button onClick={createBookmarks} id="add">
+                ADD
```
Comment (BlakeEriks) [inline src/components/PublicBookmarks.js:54]: We need an add button on every bookmark

**Review by BlakeEriks — COMMENTED** [review]
On our way to greatness!

**Pull request merged**